pkg/kv/kvserver/allocator_scorer.go

```diff
@@ -209,6 +212,18 @@ func (c candidate) compare(o candidate) float64 {
 	if c.rangeCount == 0 && o.rangeCount == 0 {
 		return 0
 	}
+	// For same node rebalances we need to manually consider the
+	// minRangeRebalanceThreshold, since in lopsided clusters their under/over
+	// replication is not properly captured in the balanceScore.
+	if c.store.Node.NodeID == o.store.Node.NodeID && c.store.Node.NodeID != 0 {
```

Comment:
I don't understand this. You shouldn't need it if we are trying to have the same number of ranges per store. If you want the same number of ranges per node - Is it not possible then to fix the balanceScore instead? We currently compute (and compare to) the mean number of ranges per store. It can change to be per node. Each store will then compare to the mean per node divided to the number of stores for its node.